Integrations/EDL/EDL.py

```diff
@@ -167,20 +208,143 @@ def find_indicators_to_limit_loop(indicator_query: str, limit: int, total_fetche
+        if (int(str(max_ip).split('.')[-1]) % 2) == 0:
+            ip_ranges.append(str(max_ip))
+            max_ip = group[-2]
+            moved_ip = True
+
+        # if both min and max ips were shifted and there are only 2 ips in the range
+        # we added both ips by the shift and now we move to the next  range
+        if moved_ip and len(group) == 2:
+            continue
+
+        else:
+            ip_ranges.append(str(iprange_to_cidrs(min_ip, max_ip)[0].cidr))
+
+    return ip_ranges
+
+
+def ip_groups_to_ranges(ip_range_groups: list):
+    """Collapse ip groups list to ranges.
+
+    Args:
+        ip_range_groups (list): a list of lists containing connected IPs
+
+    Returns:
+        list. a list of Ranges.
+    """
+    ip_ranges = []  # type:List
+    for group in ip_range_groups:
+        # handle single ips
+        if len(group) == 1:
+            ip_ranges.append(str(group[0]))
+            continue
+
+        min_ip = group[0]
+        max_ip = group[-1]
+        ip_ranges.append(str(min_ip) + "-" + str(max_ip))
+
+    return ip_ranges
+
+
+def ips_to_ranges(ips: list, collapse_ips):
+    """Collapse IPs to Ranges or CIDRs.
+
+    Args:
+        ips (list): a list of IP strings.
+        collapse_ips (str): Whether to collapse to Ranges or CIDRs.
+
+    Returns:
+        list. a list to Ranges or CIDRs.
+    """
+    ips_range_groups = []  # type:List
+    ips = sorted(ips)
+
+    if len(ips) > 0:
+        ips_range_groups.append([ips[0]])
+
+    if len(ips) > 1:
+        for ip in ips[1:]:
+            appended = False
+
+            for group in ips_range_groups:
+                if IPAddress(int(ip) + 1) in group or IPAddress(int(ip) - 1) in group:
+                    group.append(ip)
+                    sorted(group)
+                    appended = True
+
+            if not appended:
+                ips_range_groups.append([ip])
+
+    if collapse_ips == COLLAPSE_TO_RANGES:
+        return ip_groups_to_ranges(ips_range_groups)
+
+    else:
+        return ip_groups_to_cidrs(ips_range_groups)
+
+
+def create_values_for_returned_dict(iocs: list, collapse_ips: str = DONT_COLLAPSE) -> Tuple[dict, int]:
     """
     Create a dictionary for output values
     """
     formatted_indicators = []
+    ipv4_formatted_indicators = []
+    ipv6_formatted_indicators = []
     for ioc in iocs:
         value = ioc.get('value')
+        type = ioc.get('indicator_type')
         if value:
-            formatted_indicators.append(value)
-    return {EDL_VALUES_KEY: list_to_str(formatted_indicators, '\n')}
+            if collapse_ips != DONT_COLLAPSE and type == 'IP':
+                ipv4_formatted_indicators.append(IPAddress(value))
+
+            elif collapse_ips != DONT_COLLAPSE and type == 'IPv6':
+                ipv6_formatted_indicators.append(IPAddress(value))
+
+            else:
+                formatted_indicators.append(value)
+
+    if len(ipv4_formatted_indicators) > 0:
```

Comment:
we are doing this even if we want to collapse?

Reply:
no - IPs will be inserted to the list only in cases collapse is active see lines 325 and 328 for that.

But even if the collapse is active it does not mean that IPs actually exist hence this check.